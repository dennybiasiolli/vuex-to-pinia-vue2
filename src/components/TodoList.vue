<template>
  <div>
    <div v-for="todoItem of todoList" :key="todoItem.id">
      <input type="checkbox" v-model="todoItem.completed" />
      <span v-text="todoItem.name"></span>
    </div>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex'
import { GET_TODO_LIST } from '@/store/modules/todo-types'

const { mapState, mapActions } = createNamespacedHelpers('todo')

export default {
  computed: {
    ...mapState(['todoList', 'todoListLoading', 'todoListLoadingError']),
  },
  mounted() {
    this[GET_TODO_LIST]()
  },
  methods: {
    ...mapActions([GET_TODO_LIST]),
  },
}
</script>
