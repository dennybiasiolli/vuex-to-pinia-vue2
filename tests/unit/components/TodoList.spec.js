import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import TodoList from '@/components/TodoList.vue'
import { GET_TODO_LIST } from '@/store/modules/todo-types'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('TodoList.vue', () => {
  let store;
  let todoActions = {
    [GET_TODO_LIST]: jest.fn(),
  }
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        todo: {
          namespaced: true,
          state: {
            todoList: [
              { id: 1, name: 'test', completed: true },
              { id: 2, name: 'test2', completed: false },
            ],
            todoListLoading: false,
            todoListLoadingError: null,
          },
          actions: todoActions,
        },
      },
    })
  })
  test('should respect the snapshot', () => {
    const wrapper = shallowMount(TodoList, { localVue, store })
    expect(wrapper.element).toMatchSnapshot()
  })
})
