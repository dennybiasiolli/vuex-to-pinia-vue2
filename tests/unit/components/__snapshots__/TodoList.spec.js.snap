// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList.vue should respect the snapshot 1`] = `
<div>
  <div>
    <input
      type="checkbox"
    />
     
    <span>
      test
    </span>
  </div>
  <div>
    <input
      type="checkbox"
    />
     
    <span>
      test2
    </span>
  </div>
</div>
`;
